### ISSUE TITLE
Populating a field turns it into an array

### ISSUE THREAD

**Issue by dbronin** [Automattic/mongoose#10069]
**Do you want to request a *feature* or report a *bug*?**
Reporting a _bug_

**What is the current behavior?**
Currently, if populating a field of a certain type, the field turns into an array.

**If the current behavior is a bug, please provide the steps to reproduce.**
```javascript
'use strict'

const util = require('util')
const mongoose = require('mongoose')

mongoose.set('useFindAndModify', false)

const { Schema } = mongoose

run().catch(err => console.log(err))

async function run() {
await mongoose.connect('mongodb://localhost:27017/test', {
        useNewUrlParser: true,
        useUnifiedTopology: true
    })

const Books = mongoose.model('books', new Schema({name: String, author: Schema.Types.ObjectId}))
    const Authors = mongoose.model('authors', new Schema({name: String}))

    const anAuthor = new Authors({name: 'Author1'})
    await anAuthor.save()

    const aBook1 = new Books({name: 'Book1', author: anAuthor.id})
    await aBook1.save()
    const aBook2 = new Books({name: 'Book2', author: anAuthor.id})
    await aBook2.save()

    const populateOptions = [{
            path: 'author',
            model: 'authors',
            select: '_id name'
        }
    ]

    //Get the books and make book2 author an object with an _id instead of just an objectId
    const books = (await Books.find().lean()).map(aBook => {
        //aBook = aBook.toObject()
        if (!aBook._id.equals(aBook1.id)) {
            aBook.author = {_id: aBook.author}
        }
        return aBook
    })

    console.log('books = ' + util.inspect(books, false, null, true))
    //after some manipulation, books array:
    /*const books = [{
        _id: aBook1.id,
        name: aBook1.name
        author: anAuthor.id
    }, {
        _id: aBook2.id,
        name: aBook2.name
        author: {_id: anAuthor.id }
    }]
    */

    //now, we try to populate the array to get authors
    const populatedBooks = await Books.populate(books, populateOptions)

    console.log('populatedBooks = ' + util.inspect(populatedBooks, false, null, true))

    //after population, populatedBooks array:
    /*const populatedBooks = [{
        _id: aBook1.id,
        name: aBook1.name
        author: { _id: anAuthor.id, name: anAuthor.name }
    }, {
        _id: aBook2.id,
        name: aBook2.name
        author: [{_id: anAuthor.id, name: anAuthor.name}] //*this should NOT be an array
    }]
    */

    await mongoose.connection.dropDatabase()
}
```

**What is the expected behavior?**
After the populate, an object field should remain an object.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node.js: 12.18.3
MongoDB: 4.2.13
Mongoose: 5.12.2

Thank you!

**Comment by Duchynko** [comment]
Mmm, I've never seen someone use `populate` this way. Also couldn't find an example of `populate` being used on an object instead of an ObjectId in the documentation.

Is there any specific reason why you use it this way? I think it's rather a misuse of the method than a bug. But I might be wrong :)

**Comment by dbronin** [comment]
@Duchynko You are correct in that this may be an uncommon and maybe even an improper use case. The problem, however, is in the inconsistency. It should either not work, or work properly by not turning an object into an array. Further, this behavior is a change from older versions.

One scenario for this is that you have a partially populated array and need to populate the rest of it.

**Comment by vkarpov15** [comment]
@dbronin thanks for reporting, the fix will be in v5.12.14. This would work correctly without [lean](https://mongoosejs.com/docs/tutorials/lean.html), so it should also work with lean.

**Comment by dbronin** [comment]
@vkarpov15 Thank you so much!
